src/plugins/discover/public/application/components/json_code_editor/json_code_editor_common.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import './json_code_editor.scss';
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { monaco, XJsonLang } from '@kbn/monaco';
+import { EuiButtonEmpty, EuiCopy, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { CodeEditor } from '../../../../../kibana_react/public';
+
+const codeEditorAriaLabel = i18n.translate('discover.json.codeEditorAriaLabel', {
+  defaultMessage: 'Read only JSON view of an elasticsearch document',
+});
+const copyToClipboardLabel = i18n.translate('discover.json.copyToClipboardLabel', {
+  defaultMessage: 'Copy to clipboard',
+});
+
+interface JsonCodeEditorCommonProps {
+  jsonValue: string;
+  onEditorDidMount: (editor: monaco.editor.IStandaloneCodeEditor) => void;
+  width?: string | number;
+  hasLineNumbers?: boolean;
+}
+
+export const JsonCodeEditorCommon = ({
+  jsonValue,
+  width,
+  hasLineNumbers,
+  onEditorDidMount,
+}: JsonCodeEditorCommonProps) => {
+  return (
+    <EuiFlexGroup className="dscJsonCodeEditor" direction="column" gutterSize="s">
+      <EuiFlexItem>
+        <EuiSpacer size="s" />
```

Comment:
When initially loaded, jsonValue is `null`, this should be omitted, or the component should not be rendered in this case (return `null` early), since it would trigger an unnecessary re-rendering in this case.